pact/message_provider.py

```diff
@@ -115,22 +115,37 @@ def verify(self):
         return_code, _ = verifier.verify_pacts(pact_files, verbose=False)
         assert (return_code == 0), f'Expected returned_code = 0, actual = {return_code}'
 
-    def verify_with_broker(self, enable_pending=False, include_wip_pacts_since=None, **kwargs):
+    def verify_with_broker(
+            self,
+            *pacts,
+            enable_pending=False,
+            include_wip_pacts_since=None,
+            **kwargs
+    ):
         """Use Broker to verify.
 
-        Args:
-            broker_username ([String]): broker username
-            broker_password ([String]): broker password
-            broker_url ([String]): url of broker
-            enable_pending ([Boolean])
-            include_wip_pacts_since ([String])
-            publish_version ([String])
-
+        :param pacts: List of pact to verify. Every pact in that list can be
+            an HTTP URI or a local file.
+        :type pacts: str
+        :param enable_pending: Allow pacts which are in pending state to be
+            verified without causing the overall task to fail.
+        :type enable_pending: bool
+        :param include_wip_pacts_since: Allow pacts that don't match given
+            consumer selectors (or tags) to  be verified, without causing the
+            overall task to fail
+        :type include_wip_pacts_since: str or None
         """
-        verifier = Verifier(provider=self.provider,
-                            provider_base_url=self._proxy_url())
-
-        return_code, _ = verifier.verify_with_broker(enable_pending, include_wip_pacts_since, **kwargs)
+        verifier = Verifier(
+            provider=self.provider,
+            provider_base_url=self._proxy_url()
+        )
+
+        return_code, _ = verifier.verify_with_broker(
+            *pacts,
```

Comment:
it should be one pact url only, the idea being that the pact broker would trigger an event an in turn a webhook, to the provider build with the pact url to verify (singular).

https://docs.pact.io/pact_broker/webhooks#events

there may be multiple webhooks fired, once for the main branch of the provider and once for any deployed or released versions.

We should be clear about this or guard against it being a non uri, and not a local file.

Reply:
I'm not sure I understand what you mean. The current method signature (which I don't want to change in this pull request due this is offtopic) allows for an unlimited number of packets:

```py
def verify_pacts(self, *pacts, enable_pending=False, include_wip_pacts_since=None, **kwargs)
```

This has been the case for a long time and if it's wrong, it probably should be fixed, but it doesn't related to the current feature added. Isn't?

Reply:
would a user not call this method `pact.Verifier.verify_pacts` if using a local pact file or dir

https://github.com/pact-foundation/pact-python/blob/4e3ca3858b8343a5057de506f19a4f3284bb8bd4/pact/message_provider.py#LL110C9-L110C15

and then call this method `pact.Verifier.verify_with_broker` if they were using a broker

https://github.com/pact-foundation/pact-python/blob/4e3ca3858b8343a5057de506f19a4f3284bb8bd4/pact/message_provider.py#LL118C9-L118C27

which wants to accept a singular pact_url

Reply:
of these args, `enable_pending` & `include_wip_pacts_since` also don't make sense with regards to verifying via a broker url


        """Use Broker to verify.

        Args:
            broker_username ([String]): broker username
            broker_password ([String]): broker password
            broker_url ([String]): url of broker
            enable_pending ([Boolean])
            include_wip_pacts_since ([String])
            publish_version ([String])

        """
        
        
changed triggerd by by pact url

https://github.com/pactflow/example-provider/blob/master/src/product/product.consumerChange.pact.test.js

uses these options

```js
    const opts = {
      ...baseOpts,
      pactUrls: [process.env.PACT_URL],
      stateHandlers: stateHandlers,
      requestFilter: requestFilter
    };
```
        
        
one based on a provider change would involve a different set of parameters to a verifier

https://github.com/pactflow/example-provider/blob/master/src/product/product.providerChange.pact.test.js

```
    const fetchPactsDynamicallyOpts = {
      provider: 'pactflow-example-provider',
      consumerVersionSelectors: [
        { mainBranch: true },
        { deployed: true },
        { matchingBranch: true }
      ],
      pactBrokerUrl: process.env.PACT_BROKER_BASE_URL,
      // https://docs.pact.io/pact_broker/advanced_topics/pending_pacts
      enablePending: true,
      // https://docs.pact.io/pact_broker/advanced_topics/wip_pacts
      includeWipPactsSince: '2020-01-01'
    };

    const opts = {
      ...baseOpts,
      ...fetchPactsDynamicallyOpts,
      stateHandlers: stateHandlers,
      requestFilter: requestFilter
    };
```

There is some commonality between the modes the verifying with a broker

https://github.com/pactflow/example-provider/blob/master/src/product/pact.setup.js